lib/js/tests/gl/canvas2d_test.js

```diff
@@ -68,22 +96,96 @@ ctx.rect(0, 0, 10, 10);
 
 ctx.isPointInPath(0, 0);
 
+var linearGradient = ctx.createLinearGradient(0.0, 0.0, 0.0, 0.0);
+
+Canvas2dRe.setStrokeStyle(ctx, /* Gradient */Block.__(1, [linearGradient]));
+
+ctx.createRadialGradient(0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+
+linearGradient.addColorStop(0.0, "red");
+
+var partial_arg = document.createElement("img");
+
+List.map(function (param) {
+      return createPattern(ctx, partial_arg, function () {
+                    switch (param) {
+                      case 108828507 : 
+                          return "repeat";
+                      case 646437021 : 
+                          return "repeat-x";
+                      case 646437022 : 
+                          return "repeat-y";
+                      case -695430532 : 
+                          return "no-repeat";
+                      
+                    }
+                  }());
+    }, /* :: */[
+      /* noRepeat */-695430532,
+      /* :: */[
+        /* repeat */108828507,
+        /* :: */[
+          /* repeatX */646437021,
+          /* :: */[
+            /* repeatY */646437022,
+            /* [] */0
+          ]
+        ]
+      ]
+    ]);
+
+var measureText = ctx.measureText("foo");
+
+var width = measureText.width;
+
+ctx.fillText("foo!", 0.0, 0.0, width);
+
+ctx.strokeText("foo!", 0.0, 0.0, width);
+
+var imageData = ctx.createImageData(0.0, 0.0);
+
+ctx.createImageData(imageData);
+
+imageData.width;
+
+imageData.height;
+
+ctx.getImageData(0.0, 0.0, 0.0, 0.0);
+
+ctx.putImageData(imageData, 0.0, 0.0, undefined, undefined, undefined, undefined);
+
+ctx.putImageData(imageData, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0);
+
 ctx.font = "10px Courier";
 
 ctx.textAlign = "left";
 
 ctx.textBaseline = "top";
 
-ctx.fillText("hi", 1, 0);
+(function (eta) {
+      var param = /* None */0;
+      var param$1 = eta;
+      param$1.fillText("hi", 1, 0, param ? param[0] : undefined);
+      return /* () */0;
+    }(ctx));
```

Comment:
This is some weird code generation. cc @bobzhang (The offending declaration is `external fillText : string => x::float => y::float => maxWidth::float? => unit = "" [@@bs.send.pipe: t];`)

I think this is caused by not terminating an optional argument with a non-optional argument, confusing the compiler about whether or not to partially apply the function, but this codegen seems rather pointless. To fix it you can probably just add a unit argument, or have an "overload" `fillTextWithMaxWidth` to avoid the optional argument altogether.